src/visitPage.js

```diff
@@ -89,37 +91,12 @@ class VisitPage extends Component {
       // { completed: false, active: false, title: 'Housing', form: HousingStep },
       { completed: false, active: false, title: 'Results', form: ResultsGraph }
     ];  // end this.steps {}
-
-    this.stepProps = {
-      currentStep:  this.state.currentStep,
-      nextStep:     this.nextStep,
-      previousStep: this.previousStep,
-      storeComplex: this.storeComplex, // Maybe put these straight on state
-      storeChecked: this.storeChecked, // Maybe put these straight on state
-      saveForm:     this.saveForm,
-      pageState:    this.state
-    };
-
   };  // End constructor()
 
-  storeChecked = (e, { name, checked }, callback) => {
-    var truth = this;
-    truth.setState({ [name]: checked },
-      function () {
-       // console.log( name, checked, truth );
-       if ( callback ) { callback( truth ); }
-    });
-  }
-
-  storeComplex = (e, { name, value }, callback) => {
-    var truth = this;
-    truth.setState(
-      { [name]: value },
-      function () {
-        // console.log( truth );
-        if ( callback ) { callback( truth ); }
-      }  // This is given no arguments
-    );
+  setClientProperty = (e, data) => {
+    let propertyName = data.name
+    let value = data.value || data.checked  //This handles both complex values and checked values
```

Comment:
I believe this should be done the other way around. I know currently our check boxes don't have `value`s, but I believe that they can actually have `value`s, while other inputs shouldn't have a `checked` attribute. Unless someone else knows otherwise, we should check for `data.checked` first.

Reply:
Good point. Will change.